plugins/graphite/check-data.rb

```diff
@@ -179,6 +179,14 @@ def retrieve_data
           url_opts[:http_basic_authentication] = [config[:username], pass.chomp]
         end # we don't have both username and password trying without
 
+        no_proxy = ENV["no_proxy"]
+        unless no_proxy.nil?
+          domain_list = no_proxy.split(',')
+          if domain_list.any? { |domain| config[:server].include?(domain) }
+            url_opts[:proxy] = config[:proxy]
+          end
+        end
```

Comment:
What about having a new parameter in this check `--proxy` or similar in order to just use the proxy when that parameter is set ?
If that parameter is not set, I guess `url_opts[:proxy]` should be `nil`